src/main/java/spoon/support/JavaOutputProcessor.java

```diff
@@ -217,38 +217,32 @@ private void createModuleFile(CtModule module) {
 		}
 	}
 
-	private File getModuleFile(CtModule module) {
-		File moduleDir;
-		if (module == null || module.isUnnamedModule() || getEnvironment().getComplianceLevel() <= 8) {
-			moduleDir = new File(getOutputDirectory().getAbsolutePath());
-		} else {
-			// Create current package dir
-			moduleDir = new File(getOutputDirectory().getAbsolutePath() + File.separatorChar + module.getSimpleName());
-		}
-		if (!moduleDir.exists()) {
-			if (!moduleDir.mkdirs()) {
-				throw new RuntimeException("Error creating output directory");
-			}
-		}
-		return moduleDir;
+	private Path getElementPath(CtModule type) {
+		return createFolders(getEnvironment().getOutputDestination()
+				.getOutputPath(type, null, null));
 	}
 
-	private File getPackageFile(CtPackage pack) {
-		File packageDir;
-		if (pack.isUnnamedPackage()) {
-			packageDir = this.getModuleFile(pack.getDeclaringModule());
-		} else {
-			// Create current package dir
-			packageDir = new File(this.getModuleFile(pack.getDeclaringModule()).getAbsolutePath() + File.separatorChar + pack.getQualifiedName().replace('.', File.separatorChar));
-		}
-		if (!packageDir.exists()) {
-			if (!packageDir.mkdirs()) {
+	private Path getElementPath(CtPackage type) {
+		return createFolders(getEnvironment().getOutputDestination()
```

Comment:
Why don't you delegate the creation of the folder to the output destination strategy? We can imagine different implementation e.g. a strategy for not creating new directory, or for not overwritten existing files.

Reply:
> it is not exactly the same responsibility, what you ask is more something like, isToPrint, not whereToPrint

In fact I see the `OutputDestinationStrategy` as the main handler in Spoon to manage the pathes of the output. In that sense, its responsibility is indeed, not only to say where to print, but also to say if it's ok to print things at this specific location. 
The idea behind it is that you only need to provide your own implementation of `OutputDestinationStrategy` whenever you need a specific behaviour related to path, beeing where, if or how.

Reply:
I don't know how to communicate with the output processor. The use case is not clear to me.

Reply:
Maybe just rename `getOutputPath` to `createOutputPath`, and inside this method you create the needed folders?..